File: backend/src/main/java/com/tarnai/duelovky/games/entity/Game.java
package com.tarnai.duelovky.games.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity(name = "games")
@Getter
@Setter
public class Game {
    @Id
    @Column(name = "game_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long gameId;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

    @Column(name = "perex")
    private String perex;

    @Column(name = "description")
    private String description;

    @OneToMany(
        mappedBy = "game",
        cascade = CascadeType.ALL,
        orphanRemoval = true
    )
    @JsonManagedReference
    private List<GameInfo> gameInfo;

    @ManyToMany
    @JoinTable(
        name = "games_gamecategories",
        joinColumns = @JoinColumn(name = "game_id"),
        inverseJoinColumns = @JoinColumn(name = "game_category_id")
    )
    @JsonManagedReference
    private List<GameCategory> gameCategories;
}

File: frontend/src/types/game.ts
export enum GameInfoType {
	CONTROL = 'CONTROL',
	GENERAL = 'GENERAL'
}

export interface GameCategory {
	gameCategoryId: number;
	name: string;
}
export interface GameInfo {
	gameInfoId: number;
	gameInfo: string;
	infoLabel: GameInfoType;
}
export interface Game {
	gameId: number;
	name: string;
	perex: string;
	description: string;
	gameInfo: GameInfo[];
	gameCategories: GameCategory[];
}
export type GamePreview = Pick<Game, 'gameId' | 'name' | 'perex' | 'gameCategories'>;
